api/src/main/java/org/openmrs/module/insuranceclaims/api/service/fhir/impl/FHIREligibilityServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package org.openmrs.module.insuranceclaims.api.service.fhir.impl;
+
+import ca.uhn.fhir.util.DateUtils;
+import org.hl7.fhir.dstu3.model.EligibilityRequest;
+import org.hl7.fhir.dstu3.model.EligibilityResponse;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.openmrs.module.fhir.api.util.FHIRConstants;
+import org.openmrs.module.insuranceclaims.api.model.InsurancePolicy;
+import org.openmrs.module.insuranceclaims.api.model.InsurancePolicyStatus;
+import org.openmrs.module.insuranceclaims.api.service.fhir.FHIREligibilityService;
+
+import java.util.Date;
+
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.IdentifierUtil.buildReference;
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.InsuranceClaimConstants.CONTRACT_EXPIRE_DATE_ORDINAL;
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.InsuranceClaimConstants.CONTRACT_POLICY_ID_ORDINAL;
+
+public class FHIREligibilityServiceImpl implements FHIREligibilityService {
+
+    @Override
+    public EligibilityRequest generateEligibilityRequest(String  policyId) {
+        EligibilityRequest request = new EligibilityRequest();
+        Reference patient = buildReference(FHIRConstants.PATIENT, policyId);
+        request.setPatient(patient);
+
+        return request;
+    }
+
+    @Override
+    public InsurancePolicy generateInsurancePolicy(EligibilityResponse response) {
+        Reference contract = response.getInsuranceFirstRep().getContract();
+        InsurancePolicy policy = new InsurancePolicy();
+
+        Date expireDate = getExpireDateFromContractReference(contract);
+        policy.setExpiryDate(expireDate);
+        policy.setStatus(getPolicyStatus(policy));
+        policy.setPolicyNumber(getPolicyIdFromContractReference(contract));
+
+        return policy;
+    }
+
+    public String getPolicyIdFromContractReference(Reference contract) {
```

Comment:
is this required to be public ? if yes please remember about override annotation